src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1309,85 +1311,169 @@ public static unsafe int LastIndexOfAny(ref byte searchSpace, byte value0, byte
+                    offset = (nint)length - sizeof(uint);
+                    // Compare start
+                    if (LoadUInt(ref first) != LoadUInt(ref second))
                     {
-                        if (LoadVector(ref first, offset) != LoadVector(ref second, offset))
-                        {
-                            goto NotEqual;
-                        }
-                        offset += Vector<byte>.Count;
+                        goto NotEqual;
                     }
-                    return LoadVector(ref first, lengthToExamine) == LoadVector(ref second, lengthToExamine);
                 }
 
-                Debug.Assert((byte*)lengthToExamine >= (byte*)sizeof(UIntPtr));
-
-                lengthToExamine -= sizeof(UIntPtr);
-                while ((byte*)lengthToExamine > (byte*)offset)
+                // Compare end
+                return LoadUInt(ref first, offset) == LoadUInt(ref second, offset);
+            }
+#endif
+            if (length >= sizeof(ushort))
+            {
+                nint offset = 0;
+                if (length > sizeof(ushort))
                 {
-                    if (LoadUIntPtr(ref first, offset) != LoadUIntPtr(ref second, offset))
+                    // Set offset for next compare to sizeof(ushort) from end
+                    offset = (nint)length - sizeof(ushort);
+                    // Compare start
+                    if (LoadUShort(ref first) != LoadUShort(ref second))
                     {
                         goto NotEqual;
                     }
-                    offset += sizeof(UIntPtr);
                 }
-                return LoadUIntPtr(ref first, lengthToExamine) == LoadUIntPtr(ref second, lengthToExamine);
+
+                // Compare end
+                return LoadUShort(ref first, offset) == LoadUShort(ref second, offset);
             }
 
-            Debug.Assert((byte*)lengthToExamine < (byte*)sizeof(UIntPtr));
+            if (length != 0)
+            {
+                // Only length 1 possible
+                Debug.Assert((int)length == 1);
+                return first == second;
+            }
 
-            // On 32-bit, this will never be true since sizeof(UIntPtr) == 4
-#if TARGET_64BIT
-            if ((byte*)lengthToExamine >= (byte*)sizeof(int))
+            Debug.Assert((int)length == 0);
+            goto Equal;
+        Longer:
+            // Only check that the ref is the same if buffers are large, and hence
+            // its worth avoiding doing unnecessary comparisons
+            if (Unsafe.AreSame(ref first, ref second))
             {
-                if (LoadInt(ref first, offset) != LoadInt(ref second, offset))
+                // This becomes a conditional jmp foward to not favor it. (See comment at "Equal:" label)
+                return true;
+            }
+
+            if (Sse.IsSupported)
+            {
+                if (Avx.IsSupported && (nint)length >= Vector256<byte>.Count)
```

Comment:
Should instead read `length >= (nuint)Vector256<byte>.Count`? Since _length_ is coming in as an unsigned integer, we should avoid comparing it as if it were a signed integer. Large values could overflow and cause problems. So _offset_, _lengthToExamine_, and other locals should ideally also be unsigned.

Reply:
Done